### PR TITLE
Adding default main.tf for Teamcity usage

### PR DESCRIPTION
**Pull request by cgawron1**
main.tf from teamcity parameters used by DCOS Integration Tests

## High-level description

What features does this change enable? What bugs does this change fix?

## Corresponding DC/OS tickets (required)

  - [D2IQ-ID](https://jira.d2iq.com/browse/D2IQ-ID) JIRA title / short description.

## Related tickets (optional)

  - [D2IQ-ID](https://jira.mesosphere.com/browse/D2IQ-<number>) JIRA title / short description.

## Checklist for component/package updates:

If you are changing components or packages in DC/OS (e.g. you are bumping the sha or ref of anything underneath `packages`), then in addition to the above please also include:

  - [ ] Change log from the last version integrated (this should be a link to commits for easy verification and review): [example](https://github.com/dcos/dcos-mesos-modules/compare/f6fa27d7c40f4207ba3bb2274e2cfe79b62a395a...6660b90fbbf69a15ef46d0184e36755881d6a5ae)
  - [ ] Included a test which will fail if code is reverted but test is not. If there is no test please explain.
  - [ ] Test Results: [link to CI job test results for component]
  - [ ] Code Coverage (if available): [link to code coverage report]

### CODE REVIEW

File: teamcity/main.tf [hunk 1]
```diff
@@ -0,0 +1,51 @@
+provider "aws" {
+  region = "%env.AWS_REGION%"
```
Comment (orsenthil) [inline teamcity/main.tf:2]: Here and throughout this file, I think, we will have to use the format in which terraform can recognize environment variables. 

They are referenced as `"${var.ENV_VARNAME}"`

https://stackoverflow.com/a/36672931/18852

File: teamcity/main.tf [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  value = "${module.dcos.masters-ips}"
+}
+
+output "cluster-address" {
+  value = "${module.dcos.masters-loadbalancer}"
+}
+
+output "public-agents-loadbalancer" {
+  value = "${module.dcos.public-agents-loadbalancer}"
+}
```
Comment (orsenthil) [inline teamcity/main.tf:51]: Adding a new line will be desirable.

File: teamcity/main.tf [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+  dcos_calico_network_cidr = "172.17.0.0/16"
+  dcos_variant = "open"
+
+  providers = {
+    aws = "aws"
+  }
+
+  tags = {
+    build_id = "%teamcity.build.id%"
```
Comment (orsenthil) [inline teamcity/main.tf:36]: Even this will have to provided as terraform specific environment variable.

**Comment by DominikDary** [comment]
@jeschkies fyi

**Comment by jeschkies** [comment]
IMHO, this should be similar to `test_util` but this is a start so I'm fine with it.

File: teamcity/main.tf [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+module "dcos" {
+  source  = "dcos-terraform/dcos/aws"
+  version = "~> 0.2.10"
+
+  cluster_name        = "tf-ci-${var.teamcity.build.id}-"
+  cluster_name_random_string = true
+
+  ssh_public_key_file = "id_rsa.pub"
+  admin_ips           = ["0.0.0.0/0"]
```
Comment (janisz) [inline teamcity/main.tf:13]: How about restricting access to teamcity node
https://github.com/dcos/dcos/blob/master/test_util/main.tf#L102

**Comment by orsenthil** [comment]
> We already have one main.tf used by teamcity. Can we merge them?

Possible. But far more refactor will be required. Since this ticket was about replacing the teamcity parameter. I suggest that we take one step at at time. 1) Replace the teamcity parameter with proper main.tf in the repo and then 2) merge the two main.tf into single one and test it.

**Comment by DominikDary** [comment]
> > We already have one main.tf used by teamcity. Can we merge them?
> 
> Possible. But far more refactor will be required. Since this ticket was about replacing the teamcity parameter. I suggest that we take one step at at time. 1) Replace the teamcity parameter with proper main.tf in the repo and then 2) merge the two main.tf into single one and test it.

I agree what @orsenthil said. Doing on step at the time definitely makes sense. 👍

File: teamcity/main.tf (hunk 4 above)
Comment (cgawron1) [reply teamcity/main.tf]: done

**Comment by cgawron1** [comment]
@mesosphere-mergebot changelog-not-required not code related, changes are for CI only

**Comment by cgawron1** [comment]
@mesosphere-mergebot sync

**Comment by orsenthil** [comment]
@mesosphere-mergebot sync

**Pull request merged**